static/app/components/sidebar/sidebarPanel.tsx

```diff
@@ -79,6 +80,13 @@ function SidebarPanel({
         return;
       }
 
+      // If we are in Sandbox, don't hide panel when the modal is clicked (before the email is added)
+      if (
+        ConfigStore.get('demoMode') &&
+        window.localStorage.getItem('addedEmail') !== '1'
```

Comment:
can we give this local storage key a better name? That's going to be pretty hard to grep for later